DerivativeCalculator/Operator.cs

```diff
@@ -776,6 +776,170 @@ out wildcards
 							}
 						}
 					}
+					/*
+							if (TreeUtils.MatchPattern(
+								new Constant(1),
+								new Pow(new Sin(new Wildcard('a')), new Constant(2))
+								, out wildcards
+							))
+							{
+								TreeNode a = wildcards['a'];
+								addToDict = false;
+								coefficientDict[otherNode] = new Pow(new Cos(a), new Constant(2));
+								coefficientDict.Remove(node);
+							}
+							*/
+							//A TESZTEK NEM MŰKÖDNEK BUTA VAGYOK
+							if (TreeUtils.MatchPattern(
+								node,
+								new Pow(
+									new Wildcard('x'),
+									new Div(
+										new Log(new Wildcard('x')),
+										new Log(new Wildcard('a'))
+									)
+								),
+								out wildcards
+							))
+							{
+								TreeNode x = wildcards['x'];
+								TreeNode a = wildcards['a'];
+
+								coefficientDict.Remove(node);
+
+								coefficientDict[a] = x;
+								
+
+								addToDict = false;
+								break;
+							}
+
```

Comment:
ez jó, viszont fordítva lenne érdemes checkolni, mert egyszerűsíteni szeretnénk
that said, szerintem fordítva nem nagyon kerülne elő a vadonban